### ISSUE TITLE
Additional Modules

### ISSUE THREAD

**Issue by shishkin** [SingaporeScalaProgrammers/scala-workshop#6]
Quoting @damien-neveu from #5:

>Then we need to decide what the following modules will be about : how about independent modules about different aspects of the language (e.g. Arrays, Maps and Tuples, Collections, Classes, Pattern Matching with Case Classes...etc). Each module could be an independent presentation with a code-base/test-suite. Or each module could build onto the previous one to complete a bigger project, say a REST API ?

**Comment by shishkin** [comment]
How about another tutorial where participants build something more substantial (REST API might be a good example) while learning collections, tuples, case classes and pattern matching? Building something, even while barely scratching the language surface, might be more motivating than going over language features one by one. I think sparking the interest to continue learning Scala after the workshop is the key, and to do that we need to show practical examples. Thoughts?

**Comment by hhandoko** [comment]
Focusing on working project to project sounds like a good idea. How about keeping to console / command line apps, but on increasingly complex problem or look at data processing (e.g. ETL from CSV data).

I feel like going straight to to REST API will be quite a jump. There needs to be some additional materials to explain conventions on the framework used: `akka-http`, `playframework`, or `scalatra`, etc.

**Comment by damien-neveu** [comment]
@shishkin Agreed with building something that covers several aspects of the language.
@hhandoko True a REST API requires additional dependencies : akka-http may look esoteric to complete beginners and play-framework is a bit heavy. Transforming a CSV sounds like a good idea.
